### PR TITLE
Hardening phase SonarQube 6 - Daniel 2017-10-13

### PR DESCRIPTION
**Pull request by bartfastiel**

### CODE REVIEW

File: sonar-plugin-api/src/main/java/org/sonar/api/CoreProperties.java [hunk 1]
```diff
@@ -269,7 +269,9 @@
 
   /**
    * @since 2.11
+   * @deprecated since 6.7
    */
+  @Deprecated
   String CPD_CROSS_PROJECT = "sonar.cpd.cross_project";
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/CoreProperties.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)

File: server/sonar-server/src/main/java/org/sonar/server/computation/task/projectanalysis/duplication/IntegrateCrossProjectDuplications.java [hunk 2]
```diff
@@ -60,6 +61,8 @@
   public IntegrateCrossProjectDuplications(Configuration config, DuplicationRepository duplicationRepository) {
     this.config = config;
     this.duplicationRepository = duplicationRepository;
+    config.getBoolean(CoreProperties.CPD_CROSS_PROJECT).filter(s -> s).ifPresent(s ->
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/computation/task/projectanalysis/duplication/IntegrateCrossProjectDuplications.java]: MINOR Remove this use of "CPD_CROSS_PROJECT"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

File: sonar-core/src/main/java/org/sonar/core/config/CorePropertyDefinitions.java [hunk 3]
```diff
@@ -218,10 +218,12 @@ private CorePropertyDefinitions() {
       PropertyDefinition.builder(CoreProperties.CPD_CROSS_PROJECT)
```
Comment (SonarTech) [inline sonar-core/src/main/java/org/sonar/core/config/CorePropertyDefinitions.java]: MINOR Remove this use of "CPD_CROSS_PROJECT"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

**Comment by SonarTech** [comment]
SonarQube analysis reported 3 issues
* ![MINOR][MINOR] 2 minor
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: sonar-core/src/main/java/org/sonar/core/config/CorePropertyDefinitions.java [hunk 4]
```diff
@@ -218,10 +218,12 @@ private CorePropertyDefinitions() {
...
         .defaultValue(Boolean.toString(false))
         .name("Cross project duplication detection")
-        .description("By default, SonarQube detects duplications at project level. This means that a block "
+        .description("[Deprecated Feature] By default, SonarQube detects duplications at project level. This means that a block "
```
[the remark this answers is not in the document]
Comment (bartfastiel) [reply sonar-core/src/main/java/org/sonar/core/config/CorePropertyDefinitions.java]: Done. Thanks! 👍

File: server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/SearchAction.java [hunk 5]
```diff
@@ -104,7 +104,7 @@ public void define(WebService.NewController controller) {
     action
       .createParam(ComponentsWsParameters.PARAM_ORGANIZATION)
       .setDescription("Organization key. If no organization key is provided, this defaults to the organization of the specified project. If neither organization nor project are" +
-        "specified, the default organization will be used.")
+        " specified, the default organization will be used.")
```
[the remark this answers is not in the document]
Comment (bartfastiel) [reply server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/SearchAction.java]: Done.

File: server/sonar-server/src/main/java/org/sonar/server/computation/task/projectanalysis/duplication/IntegrateCrossProjectDuplications.java [hunk 6]
```diff
@@ -60,6 +61,9 @@
   public IntegrateCrossProjectDuplications(Configuration config, DuplicationRepository duplicationRepository) {
     this.config = config;
     this.duplicationRepository = duplicationRepository;
+    if (config.getBoolean(CoreProperties.CPD_CROSS_PROJECT).orElse(false)) {
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/computation/task/projectanalysis/duplication/IntegrateCrossProjectDuplications.java]: MINOR Remove this use of "CPD_CROSS_PROJECT"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

Comment (teryk) [reply server/sonar-server/src/main/java/org/sonar/server/computation/task/projectanalysis/duplication/IntegrateCrossProjectDuplications.java]: 👍

**Comment by bartfastiel** [comment]
I'm waiting for a functional verification of a product manager.

**Pull request merged**